### DISCUSSION TITLE
log  inputBlobReader instanceof Uint8Array,the console shows false

### DISCUSSION THREAD

**Discussion by hugetiny** [gildas-lormeau/zip.js#278] [Q&A] [answered]
```
class BlobReader extends Reader {

	constructor(blob) {
		super();
		this.blob = blob;
		this.size = blob.size;
	}

	async readUint8Array(offset, length) {
		const reader = new FileReader();
		return new Promise((resolve, reject) => {
			reader.onload = event => resolve(new Uint8Array(event.target.result));
			reader.onerror = () => reject(reader.error);
			reader.readAsArrayBuffer(this.blob.slice(offset, offset + length));
		});
	}
}
```
in source code:BlobReader  returns  a Uint8Array
```
const inputBlobReader = new zip.BlobReader(inputBlob)
```
but when I log  inputBlobReader instanceof Uint8Array,the console shows false
why?

**Comment by hugetiny** [comment, marked as answer]
```
const inputBlobReader = new zip.BlobReader(inputBlob);
const buffer = await inputBlobReader.readUint8Array(0,inputBlobReader.size)
console.log(buffer instanceof Uint8Array)// true
console.log(buffer.length)// 0
```

**Reply by gildas-lormeau** [reply]
Note that you shouldn't need to call `readUint8Array` in your code. This method is used internally by zip.js to retrieve data from a `Reader`.

**Reply by hugetiny** [reply]
> Note that you shouldn't need to call `readUint8Array` in your code. This method is used internally by zip.js to retrieve data from a `Reader`.

I want to generate an unique code（hash） from a blob  and that method only supports Uint8Array.And I don't want to readUint8Array twice for performance issue.

**Reply by gildas-lormeau** [reply]
That's indeed a legit use case ;) Maybe you could extend `BlobReader`.
